embedded-git/src/main/java/com/playtika/test/git/EmbeddedGitBootstrapConfiguration.java

```diff
@@ -0,0 +1,87 @@
+import static com.playtika.test.git.GitProperties.BEAN_NAME_EMBEDDED_GIT;
+
+@Slf4j
+@Configuration
+@ConditionalOnExpression("${embedded.containers.enabled:true}")
+@AutoConfigureAfter(DockerPresenceBootstrapConfiguration.class)
+@ConditionalOnProperty(name = "embedded.git.enabled", matchIfMissing = true)
+@EnableConfigurationProperties(GitProperties.class)
+public class EmbeddedGitBootstrapConfiguration {
+
+    @Bean
+    @ConditionalOnToxiProxyEnabled(module = "git")
+    ToxiproxyContainer.ContainerProxy gitContainerProxy(ToxiproxyContainer toxiproxyContainer,
+                                                        @Qualifier(BEAN_NAME_EMBEDDED_GIT) GenericContainer<?> embeddedGit,
+                                                        ConfigurableEnvironment environment,
+                                                        GitProperties gitProperties) {
+        ToxiproxyContainer.ContainerProxy proxy = toxiproxyContainer.getProxy(embeddedGit, gitProperties.getPort());
+
+        Map<String, Object> map = new LinkedHashMap<>();
+        map.put("embedded.git.toxiproxy.host", proxy.getContainerIpAddress());
+        map.put("embedded.git.toxiproxy.port", proxy.getProxyPort());
+        map.put("embedded.git.toxiproxy.proxyName", proxy.getName());
+
+        MapPropertySource propertySource = new MapPropertySource("embeddedGitToxiproxyInfo", map);
+        environment.getPropertySources().addFirst(propertySource);
+        log.info("Started Git ToxiProxy connection details {}", map);
+
+        return proxy;
+    }
+
+    @Bean(name = BEAN_NAME_EMBEDDED_GIT, destroyMethod = "stop")
+    public GenericContainer<?> embeddedGit(ConfigurableEnvironment environment,
+                                           GitProperties properties,
+                                           Optional<Network> network) {
+        GenericContainer<?> gitContainer = configureCommonsAndStart(createContainer(properties), properties, log);
+        network.ifPresent(gitContainer::withNetwork);
+        registerGitEnvironment(gitContainer, environment, properties);
+        return gitContainer;
+    }
+
+    private GenericContainer<?> createContainer(GitProperties properties) {
+        return new GenericContainer<>(ContainerUtils.getDockerImageName(properties))
+                .withExposedPorts(properties.getPort())
+                .waitingFor(new HostPortWaitStrategy());
+    }
+
+    private void registerGitEnvironment(GenericContainer<?> gitContainer,
+                                        ConfigurableEnvironment environment,
+                                        GitProperties properties) {
+        Integer mappedPort = gitContainer.getMappedPort(properties.getPort());
+        String host = gitContainer.getHost();
+
+        LinkedHashMap<String, Object> map = new LinkedHashMap<>();
+        map.put("embedded.git.port", mappedPort);
+        map.put("embedded.git.host", host);
```

Comment:
Provide in log message full path how to connect to repository.

example : https://github.com/PlaytikaOSS/testcontainers-spring-boot/blob/452fbae3cf5832addbd4104607aaa1ea92acd551/embedded-mysql/src/main/java/com/playtika/test/mysql/EmbeddedMySQLBootstrapConfiguration.java#L90-L92

Reply:
done